core/shared/src/main/scala/zio/prelude/NonEmptyMap.scala

```diff
@@ -0,0 +1,224 @@
+package zio.prelude
+
+/*
+ * Copyright 2020-2022 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import zio.NonEmptyChunk
+
+import scala.language.implicitConversions
+
+/**
+ * A non-empty wrapper for the scala immutable map. Note - this does not attempt to implement all features of
+ * map but what the author considers to be the "normal ones".
+ */
+case class NonEmptyMap[K, V] private (private val map: Map[K, V]) { self =>
```

Comment:
I think this shouldn't be a `case class`, just a mere `class`, as are the other `NonEmpty*` classes

Reply:
ok, these are subtle differences the impact of which I don't really undersand. But yes, conformity to the pattern is a definite plus